presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -109,6 +114,12 @@ public long getStartTime()
     }
 
     @Override
+    public <T> T getSystemProperty(String name, Class<T> type)
+    {
+       return type.cast(systemProperties.get(name));
```

Comment:
Actually, you should cherry pick https://github.td.teradata.com/center-for-hadoop/presto-re2/commit/0a75cb9abd45cbc67a98d4f88602e3473881af67